kie-server-parent/kie-server-services/kie-server-services-openshift/src/test/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepositoryRegularTest.java

```diff
@@ -0,0 +1,350 @@
+        Resource<ConfigMap, DoneableConfigMap> configMapResource = client.configMaps().inNamespace(testNamespace)
+                                                                         .withName(TEST_KIE_SERVER_ID);
+    
+        ConfigMap configMap = configMapResource.get();
+        assertEquals(TEST_KIE_SERVER_ID, configMap.getMetadata().getName());
+    
+        Map<String, String> data = configMap.getData();
+    
+        // Avoid attribute name having '.' as it confuses jsonpath
+        String srvStateInXML = data.get(KieServerStateCloudRepository.CFG_MAP_DATA_KEY);
+        KieServerState kieServerState = (KieServerState) xs.fromXML(srvStateInXML);
+    
+        assertNotNull(kieServerState);
+        assertEquals(TEST_KIE_SERVER_ID,
+                     kieServerState.getConfiguration().getConfigItem(KieServerConstants.KIE_SERVER_ID).getValue());
+    
+    }
+
+    @Test
+    public void testStoreAndLoad() throws InterruptedException {
+        // Retrieve the seeded KSSConfigMap and Store it under new name
+        String srvStateInXML = client.configMaps().inNamespace(testNamespace)
+                                     .withName(TEST_KIE_SERVER_ID).get().getData().get(KieServerStateCloudRepository.CFG_MAP_DATA_KEY);
+        KieServerState kieServerState = (KieServerState) xs.fromXML(srvStateInXML);
+    
+        assertNotNull(kieServerState);
+        
+        /**
+         * At normal situation, DC will not be null otherwise there will no KieServer Pod running
+         */
+        createDummyDC();
+        repo.store(TEST_KIE_SERVER_ID, kieServerState);
+        assertNotNull(client.configMaps().inNamespace(testNamespace)
+                            .withName(TEST_KIE_SERVER_ID).get());
+    
+        KieServerState kssLoaded = repo.load(TEST_KIE_SERVER_ID);
+        assertNotNull(kssLoaded);
+        assertEquals(TEST_KIE_SERVER_ID,
+                     kssLoaded.getConfiguration().getConfigItem(KieServerConstants.KIE_SERVER_ID).getValue());
+    
+        KieContainerResource[] kcr = kssLoaded.getContainers()
+                                              .<KieContainerResource> toArray(new KieContainerResource[]{});
+    
+        assertEquals(2, kcr.length);
+        assertEquals("mortgages_1.0.0-SNAPSHOT", kcr[0].getContainerId());
+        assertEquals("mortgage-process_1.0.0-SNAPSHOT", kcr[1].getContainerId());
+    }
+
+    @Test
+    public void testStoreAndLoadWithRolloutTrigger() throws InterruptedException {
+        // Retrieve the seeded KSSConfigMap and Store it under new name
+        String srvStateInXML = client.configMaps().inNamespace(testNamespace)
+                                     .withName(TEST_KIE_SERVER_ID).get().getData().get(KieServerStateCloudRepository.CFG_MAP_DATA_KEY);
+        KieServerState kieServerState = (KieServerState) xs.fromXML(srvStateInXML);
+    
+        repo.store(TEST_KIE_SERVER_ID, kieServerState);
+        assertNotNull(client.configMaps().inNamespace(testNamespace)
```

Comment:
I suppose that there should be "assertTrue" here.

Reply:
Fixed the test assertion.